src/logic/safe/api/fetchTokenCurrenciesBalances.ts

```diff
@@ -109,28 +107,22 @@ async function fetchTokensWithRetriesAndAlternate(address: string) {
   return []
 }
 
-interface PriceInfo {
-  symbol: string
-  derivedCUSD: number
+type TokenPrices = {
+  [key: string]: {
+    [currency: string]: number
+  }
 }
 
-const tokensQuery = `query { tokens(number_gte: 11749000) {
-  derivedCUSD
-  symbol
-}}`
-
-async function fetchPrices(): Promise<Record<string, number>> {
+async function fetchPrices(tokens: BlockScoutTokenBalance[], selectedCurrency: string): Promise<TokenPrices> {
+  const addressesString = tokens.reduce((addresses, token) => {
+    return `${addresses},${token.contractAddress}`
+  }, '')
+  const priceUrl = `https://api.coingecko.com/api/v3/simple/token_price/celo?vs_currencies=${selectedCurrency}&contract_addresses=${addressesString}`
```

Comment:
hmm this means that coingeck has to either have the version deployed to celo or the contract address on celo must be the same as the one on Eth mainnet (or wherever the token is native to)

Reply:
I do not understand this comment